### PR TITLE
Sign and notarize macos app and installer

### PR DESCRIPTION
**Pull request by Thykof**
## Delivery Quality Checklist

- [ ] **Breaking Changes in API:**
  Does this PR introduce breaking changes in the API?
    - If yes, have you considered making it backward compatible?
    - If backward compatibility is not considered, set the "breaking-change" label.

- [ ] **Changelog:**
  - [ ] For bugfix PR, set the "bugfix" label
  - [ ] If this change should not appear in changelog, use "ignore-for-changelog" label

- [ ] **Version Update Handling:**
  Have you ensured that the version update by user is handled correctly?

- [ ] **PR Dependency:**
  Does this PR depend on another PR?
    - If yes, is it necessary for the dependency to be released prior to merging this one?

- [ ] **Documentation:**
  - [ ] Are any necessary changes made to user-facing documentation?
  - [ ] Confirm that API documentation is updated with any relevant changes.
  - [ ] Check that README and other documentation files are accurate and current.

### CODE REVIEW

File: installer/macos/create_pkg.sh [hunk 1]
```diff
@@ -7,9 +7,12 @@ set -e
...
-MASSASTATION_INSTALLER_NAME=MassaStation.app
+MASSASTATION_APPLICATION_NAME=MassaStation.app
 MASSASTATION_BINARY_NAME=massastation
 
+APPLE_DEVELOPER_ID_APPLICATION=$2
+APPLE_DEVELOPER_ID_INSTALLER=$3
+
 HOMEBREW_INSTALL_SCRIPT_URL=https://raw.githubusercontent.com/massalabs/homebrew.sh/master/homebrew-3.3.sh
 
 LICENSE_FILE_NAME=MassaStation_ToS.rtf
```
Comment (thomas-senechal) [inline installer/macos/create_pkg.sh:18]: I can't comment on the `display_usage` function so I'm commenting here. 
I think we need to update it to add the App Dev ID and Installer Dev ID

**Review by thomas-senechal — CHANGES_REQUESTED** [review]
Did you tried to run the `create_pkg.sh` script from your computer without the Dev IDs ? 
What happens in that case ? 

If packaging fails or if the installer does not work, I think we need to add an handling of building with and without the Dev IDs. Otherwise, we won't be able to build from our computers and will require to push, wait for 15min, to then check if it's working. And it's not a viable way of working IMO

**Comment by Thykof** [comment]
> Did you tried to run the `create_pkg.sh` script from your computer without the Dev IDs ? What happens in that case ?
> 
> If packaging fails or if the installer does not work, I think we need to add an handling of building with and without the Dev IDs. Otherwise, we won't be able to build from our computers and will require to push, wait for 15min, to then check if it's working. And it's not a viable way of working IMO

I modifier the script, you can now run `./macos/create_pkg.sh amd64 "Developer ID Application: Massa Labs (9X8NWA3JHQ)" "Developer ID Installer: Massa Labs (9X8NWA3JHQ)"` or `./macos/create_pkg.sh amd64`

File: installer/macos/create_pkg.sh [hunk 2]
```diff
@@ -38,20 +48,31 @@ install_massastation_build_dependencies() {
...
+    fyne package -icon ../../int/systray/embedded/logo.png -name MassaStation -appID com.massalabs.massastation -src ../cmd/massastation || fatal "fyne package failed for $MASSASTATION_APPLICATION_NAME"
+    chmod +x $MASSASTATION_APPLICATION_NAME || fatal "failed to chmod $MASSASTATION_APPLICATION_NAME"
 }
 
 # Build the package using pkgbuild.
 package() {
-    pkgbuild --component $MASSASTATION_INSTALLER_NAME --identifier com.massalabs.massastation --version $PKGVERSION \
+    # sign the application if we have a developer id
+    if [[ -n "$APPLE_DEVELOPER_ID_APPLICATION" ]]; then
+        codesign --force --options runtime --sign "$APPLE_DEVELOPER_ID_APPLICATION" $MASSASTATION_APPLICATION_NAME
```
Comment (thomas-senechal) [inline installer/macos/create_pkg.sh:63]: Why `--force` ? 🤔 

(non blocking)

**Review by thomas-senechal — APPROVED** [review]
Good job ! 
Only a non blocking comment

**Review by gregLibert — APPROVED** [review]

File: installer/macos/create_pkg.sh (hunk 2 above)
Comment (Thykof) [reply installer/macos/create_pkg.sh:63]: to replace any signature already there, it fails if there is already a signature and we don't use the --force option. However, there should not be a signature already but let's be sure.

**Pull request merged**